app/services/reports/csv_report_service.rb

```diff
@@ -0,0 +1,38 @@
+class Reports::CsvReportService
+  delegate :csv, to: :@csv_service
+
+  def initialize(records)
+    record_type = detect_record_type(records)
+    @csv_service = csv_service_class(record_type).new(records)
+  end
+
+private
+
+  class NilCsvService
+    def initialize(_records); end
+
+    def csv
+      ""
+    end
+  end
+
+  def csv_service_class(record_type)
+    return NilCsvService if record_type.nil?
+
+    "Reports::#{record_type.capitalize}CsvReportService".constantize
+  end
```

Comment:
I fear this is might make it more difficult to understand than it makes it easier to maintain.